graph/src/hypercert-minter-v-0.ts

```diff
@@ -0,0 +1,153 @@
+  // - contract.getRoleAdmin(...)
+  // - contract.hasRole(...)
+  // - contract.impactScopes(...)
+  // - contract.isApprovedForAll(...)
+  // - contract.proxiableUUID(...)
+  // - contract.rights(...)
+  // - contract.supportsInterface(...)
+  // - contract.totalSupply(...)
+  // - contract.uri(...)
+  // - contract.version(...)
+  // - contract.workScopes(...)
+}
+
+export function handleApprovalForAll(event: ApprovalForAll): void {}
+
+export function handleBeaconUpgraded(event: BeaconUpgraded): void {}
+
+export function handleImpactClaimed(event: ImpactClaimed): void {
+  let entity = new Hypercert(event.params.id.toString());
+  entity.claimHash = event.params.claimHash;
+  const contributors = [] as string[];
+  for (let i=0; i<event.params.contributors.length; i++)
+  {
+    const address = event.params.contributors[i];
+    contributors.push(address.toHexString());
+  }
+  entity.contributors = contributors;
+  entity.impactDateFrom = event.params.impactTimeframe[0];
+  entity.impactDateTo = event.params.impactTimeframe[1];
+  entity.impactScopes = event.params.impactScopes;
+  entity.rights = event.params.rights;
+  entity.uri = event.params.uri;
+  entity.version = event.params.version;
+  entity.workDateFrom = event.params.workTimeframe[0];
+  entity.workDateTo = event.params.workTimeframe[1];
+  entity.workScopes = event.params.workScopes;
+  entity.save();
+}
+
+export function handleImpactScopeAdded(event: ImpactScopeAdded): void {
+  let entity = new ImpactScope(event.params.id);
+  entity.text = event.params.text;
+  entity.save();
+}
+
+export function handleInitialized(event: Initialized): void {}
+
+export function handleRoleAdminChanged(event: RoleAdminChanged): void {}
+
+export function handleRoleGranted(event: RoleGranted): void {}
+
+export function handleRoleRevoked(event: RoleRevoked): void {}
+
+export function handleRightAdded(event: RightAdded): void {
+  let entity = new Right(event.params.id);
+  entity.text = event.params.text;
+  entity.save();
+}
+
+export function handleWorkScopeAdded(event: WorkScopeAdded): void {
+  let entity = new WorkScope(event.params.id);
+  entity.text = event.params.text;
+  entity.save();
+}
+
+export function handleTransferBatch(event: TransferBatch): void {}
+
+export function handleTransferSingle(event: TransferSingle): void {
+  const tokenId = event.params.id.toString();
+  const from = event.params.from.toHexString();
+  const to = event.params.to.toHexString();
+  const fromId = `${from}-${tokenId}`;
+  const balanceFrom = HypercertBalance.load(fromId);
+  if (balanceFrom) {
+    balanceFrom.amount = balanceFrom.amount - event.params.value;
+    balanceFrom.save();
+  }
+
+  const toId = `${to}-${tokenId}`;
+  let balanceTo = HypercertBalance.load(toId);
+  if (!balanceTo) {
+    balanceTo = new HypercertBalance(toId);
+    balanceTo.token = tokenId;
+    balanceTo.amount = BigInt.fromI32(0);
+  }
+
+  balanceTo.amount = balanceTo.amount + event.params.value;
```

Comment:
I think this should be `balanceTo.amount.plus(event.params.value)`. Same for other places where `BigInt`s are being added or substracted.

Reply:
updated